nucypher/cli/actions.py

```diff
@@ -108,6 +109,47 @@ def unlock_nucypher_keyring(emitter, password: str, character_configuration: Cha
         raise character_configuration.keyring.AuthenticationFailed
 
 
+def load_static_nodes(domains: Set[str], filepath: Optional[str] = None) -> Dict[str, 'Ursula']:
+    """
+    Non-invasively read teacher-uris from a JSON configuration file keyed by domain name.
+    and return a filtered subset of domains and teacher URIs as a dict.
+    """
+
+    if not filepath:
+        filepath = os.path.join(DEFAULT_CONFIG_ROOT, 'static-nodes.json')
+    try:
+        with open(filepath, 'r') as file:
+            static_nodes = json.load(file)
+    except FileNotFoundError:
+        return dict()   # No static nodes file, No static nodes.
+    except JSONDecodeError:
+        raise RuntimeError(f"Static nodes file '{filepath}' contains invalid JSON.")
+    filtered_static_nodes = {domain: uris for domain, uris in static_nodes.items() if domain in domains}
+    return filtered_static_nodes
+
+
+def aggregate_seednode_uris(domains: set, highest_priority: Optional[List[str]] = None) -> List[str]:
+
+    # Read from the disk
+    static_nodes = load_static_nodes(domains=domains)
+
+    # Priority 1 - URI passed via --teacher
+    uris = highest_priority or list()
```

Comment:
Out of curiosity - should this be a set i.e. may not matter, but can a node be part of more than one domain?

Reply:
No such thing as an "ordered set".  It's true that a valid code path includes a node twice, if its specified as part of two domains.  This may indicate that the overall approach can be better implemented by not storing static nodes in buckets of unique domains.

Instead of static-nodes.json :
```
STATIC_NODES = {'domain-name': 'https://uri:9151'}
```

We can do `static-nodes.conf`:
``` 
https://uri:9151 domian1
https://uri:9151 domian1, domain2, domain3
https://uri:9151 domian1, domain2
```

What do you think?

Reply:
Can the same node be serving for two different domains?
Do domains mean different testnets, federated nets, mainnet?

Reply:
You could do an ordered dict which, while redundant, acts as an ordered set. Something like:

```python
uris = OrderedDict()
...
uris['https://uri:9151'] = <some_value, eg. 'https://uri:9151'>
```

@KPrasch, your `static-nodes.conf` idea is good, but maybe a tad less efficient since all uris would need to be iterated over regardless of whether they apply to the set of domains or not...? wdyt?
Also, @michwill 's question above is pertinent - is this even a possible scenario?

Reply:
> Can the same node be serving for two different domains?
Yes.

> Do domains mean different testnets, federated nets, mainnet? 
Yes, but not necessarily.  They are domains of node discovery.

Reply:
> ... you could just leave the code as is...wdyt?

Yep - reasonable and allows for future iteration.

Also, @michwill 's question above is pertinent - is this even a possible scenario?

Yes - See above comment.